### PR TITLE
fix(frontend): incorrect errormessage for wrong contribution link

### LINKED ISSUE #2897: 🐛 [Bug] incorrect errormessage for wrong contribution-link
## 🐛 Bugreport

Wenn man eine beliebige Nummer beim Contribution-Link eingibt, bekommt man folgende Anzeige.

## 🤖 Todo
- [ ] Default text "redeem 123,45" to "The link is unknown."
- [ ] Redeem button disabled in default

### PR DESCRIPTION
**Pull request by ogerly**
## 🍰 Pullrequest
error message for incorrect redemption link,
button turned off if not correct

### Issues
- fixes #2897 

Bildschirmfoto vom 2023-04-27 10-05-46
Bildschirmfoto vom 2023-04-27 09-59-41
Bildschirmfoto vom 2023-04-27 09-59-16

### CODE REVIEW

File: frontend/src/pages/TransactionLink.vue [hunk 1]
```diff
@@ -67,13 +69,15 @@ export default {
...
           this.linkData = result.data.queryTransactionLink
           if (this.linkData.__typename === 'ContributionLink' && this.$store.state.token) {
             this.mutationLink(this.linkData.amount)
           }
         })
-        .catch((err) => {
-          this.toastError(err.message)
+        .catch(() => {
+          this.toastError(this.$t('gdd_per_link.redeemlink-error'))
+          // this.$router.push('/overview')
```
Comment (mahula) [inline frontend/src/pages/TransactionLink.vue:80]: Please remove the comment.

File: frontend/src/components/LinkInformations/RedeemInformation.vue [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
   <div class="redeem-information">
     <b-jumbotron bg-variant="muted" text-variant="dark" border-variant="info">
-      <h1 v-if="isContributionLink">
+      <h1 v-if="amount === ''">{{ $t('gdd_per_link.redeemlink-error') }}</h1>
+      <h1 v-if="isContributionLink && amount !== ''">
         {{ CONFIG.COMMUNITY_NAME }}
         {{ $t('contribution-link.thanksYouWith') }} {{ amount | GDD }}
       </h1>
-      <h1 v-else>
+      <h1 v-if="!isContributionLink && amount !== ''">
```
Comment (mahula) [inline frontend/src/components/LinkInformations/RedeemInformation.vue:9]: What about putting the conditions (amount size, kind of link) in a method instead?

Comment (mahula) [inline frontend/src/components/LinkInformations/RedeemInformation.vue:9]: What about putting the conditions (amount size, kind of link) in a method instead?

**Review by mahula — CHANGES_REQUESTED** [review]

**Review by ulfgebhardt — COMMENTED** [review]
I can say this works - tested locally

File: frontend/src/pages/TransactionLink.vue (hunk 1 above)
Comment (ogerly) [reply frontend/src/pages/TransactionLink.vue:80]: removed

File: frontend/src/components/LinkInformations/RedeemInformation.vue (hunk 2 above)
Comment (ogerly) [reply frontend/src/components/LinkInformations/RedeemInformation.vue:9]: at this point only data is transmitted, checked and displayed. 
the data is in the props and will never change. 
additional code would complicate things unnecessarily in my view. 
methods must then also be tested again.

Comment (mahula) [reply frontend/src/components/LinkInformations/RedeemInformation.vue:9]: Well, then there is no need to change this.

**Review by mahula — APPROVED** [review]

**Pull request merged**